src/controllers/controllerdebug.h

```diff
@@ -23,10 +25,24 @@ class ControllerDebug {
         s_enabled = false;
     }
 
+    static void enableTesting() {
+        s_enabled = true;
+        s_testing = true;
+    }
+
+    static ControlFlags shouldAssertForInvalidControlObjects() {
+        if (s_enabled && !s_testing) {
+            return ControlFlag::None;
```

Comment:
None means assert all assumptions. What is the reasoning that we don't want that during tests? Is this correct: 
|     | s_testing = true |  s_testing = false
|----|-----|----
| s_enabled = true |ignore |  assert |
| s_enabled = false |ignore | ignore  |

Please add a comment why.  
Thank you.

Reply:
We may also consider to rename ControlFlag::None to 
ControlFlag::AssertAll; Or AssertExeptions:None or such.

Reply:
> Please add a comment why.

This was done because some tests are expected to fail. I just pushed the new commits that disable controller testing for these tests.